src/s2wasm.h

```diff
@@ -370,22 +378,59 @@ class S2WasmBuilder {
     s = inputStart;
     while (*s) {
       skipWhitespace();
-      s = strstr(s, ".type");
-      if (!s) break;
-      mustMatch(".type");
-      Name name = getCommaSeparated();
-      skipComma();
-      if (!match("@function")) continue;
-      if (match(".hidden")) mustMatch(name.str);
-      mustMatch(name.str);
-      if (match(":")) {
-        info->implementedFunctions.insert(name);
-      } else if (match("=")) {
-        Name alias = getAtSeparated();
-        mustMatch("@FUNCTION");
-        info->aliasedFunctions.insert({name, alias});
+
+      // add function definitions and aliases
+      if (match(".type")) {
+        Name name = getCommaSeparated();
+        skipComma();
+        if (!match("@function")) continue;
+        if (match(".hidden")) mustMatch(name.str);
+        mustMatch(name.str);
+        if (match(":")) {
+          info->implementedFunctions.insert(name);
+        } else if (match("=")) {
+          Name alias = getAtSeparated();
+          mustMatch("@FUNCTION");
+          info->aliasedSymbols.insert({name, LinkerObject::SymbolAlias(alias, LinkerObject::Relocation::kFunction, 0, 0)});
+        } else {
+          abort_on("unknown directive");
+        }
+      // add data aliases
       } else {
-        abort_on("unknown directive");
+        Name lhs = getStrToSep();
```

Comment:
1. OK; in that case we should also add a test case (to alias.s is fine) to cover this behavior.
2. I don't  think it should be necessary if alias size isn't used. IIRC you are right and it doesn't matter except for allocation, and of course only the aliasee is allcoated.